pkg/runner/expression.go

```diff
@@ -130,6 +132,83 @@ func (ee expressionEvaluator) evaluate(in string, isIfExpression bool) (interfac
 	return evaluated, err
 }
 
+func (ee expressionEvaluator) EvaluateYamlNode(node *yaml.Node) error {
+	if node.Kind == yaml.ScalarNode {
+		var in string
+		if err := node.Decode(&in); err != nil {
+			return err
+		}
+		if !strings.Contains(in, "${{") || !strings.Contains(in, "}}") {
+			return nil
+		}
+		expr, _ := rewriteSubExpression(in, false)
+		if in != expr {
+			log.Debugf("expression '%s' rewritten to '%s'", in, expr)
+		}
+		res, err := ee.evaluate(in, false)
+		if err != nil {
+			return err
+		}
+		return node.Encode(res)
+	} else if node.Kind == yaml.MappingNode {
+		var m map[interface{}]yaml.Node
+		if err := node.Decode(&m); err != nil {
+			return err
+		}
+		// GitHub has this undocumented feature to merge maps, called insert directive
+		insertDirective := regexp.MustCompile(`\${{\s*insert\s*}}`)
```

Comment:
Sample usage of `${{insert}}`, key collisions result in template error on github and actions/runner
```yaml
on: push
jobs:
  a:
    strategy:
      matrix:
        a:
        - env:
              key1: val1
              ${{insert}}:
                 key2: val2
              ${{insert}}: ${{fromJSON('{"key3": "val3"}')}}
    runs-on: ubuntu-latest
    steps:
    - run: exit 0
```